pkg/common/cns-lib/volume/manager.go

```diff
@@ -1275,6 +1283,15 @@ func (m *defaultManager) deleteVolumeWithImprovedIdempotency(ctx context.Context
 	volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 	if volumeOperationRes.Fault != nil {
 		faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+
+		// If volume is not found on host, then is present in CNS DB, we will get NotFound fault.
+		// Send back success as the volume is already deleted.
+		_, isNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.NotFound)
```

Comment:
Can't we use extracted fault type set in the `faultType` here?
make sure we do not run into Panic if fault.fault is nil